Source/WorldActorRegistry/Private/WARWorldActorRegistry.cpp

```diff
@@ -0,0 +1,81 @@
+#include "WorldActorRegistry/Public/WARWorldActorRegistry.h"
+
+TArray< AActor * > UWARWorldActorRegistry::GetActorsFromClass( UClass * actor_class ) const
+{
+    TArray< AActor * > result;
+
+    if ( const auto * existing_actors = Registry.Find( actor_class ) )
+    {
+        ( *existing_actors ).GenerateValueArray( result );
+    }
+
+    return result;
+}
+
+AActor * UWARWorldActorRegistry::GetActorFromClass( UClass * actor_class ) const
+{
+    const auto existing_actors = GetActorsFromClass( actor_class );
+
+    if ( existing_actors.Num() > 0 )
+    {
+        return existing_actors[ 0 ];
+    }
+
+    return nullptr;
+}
+
+AActor * UWARWorldActorRegistry::GetActorFromClassWithTag( UClass * actor_class, const FGameplayTag gameplay_tag ) const
+{
+    const auto * existing_actors = Registry.Find( actor_class );
+
+    if ( const auto * actor = existing_actors->Find( gameplay_tag ) )
+    {
+        return *actor;
+    }
+
+    return nullptr;
+}
+
+bool UWARWorldActorRegistry::AddActorToRegistry( AActor * actor, const FGameplayTag tag )
+{
+    if ( actor == nullptr ||
+         !actor->HasAuthority() ||
+         Registry.Num() == 0 )
+    {
+        return false;
+    }
+
+    auto & existing_actors = Registry.FindOrAdd( actor->GetClass() );
+
+    if ( existing_actors.Find( tag ) )

```

Comment:
I would add an ensure. It's better to warn the user that the registration failed

Reply:
Done.